test/types/pino.test-d.ts

```diff
@@ -10,10 +10,61 @@ const error = log.error;
 
 info("hello world");
 error("this is at error level");
-info("the answer is %d", 42);
+
+// primative types
+info('simple string');
+info(true)
+info(42);
+info(3.14);
+info(null);
+info(undefined);
+
+// object types
+info({ a: 1, b: '2' });
+info(new Error());
+info(new Date());
+info([])
+info(new Map());
+info(new Set());
+
+// placeholder messages
+info('Hello %s', 'world');
+info('The answer is %d', 42);
+info('The object is %o', { a: 1, b: '2' });
+info('The json is %j', { a: 1, b: '2' });
+info('The object is %O', { a: 1, b: '2' });
+info('The answer is %d and the question is %s with %o', 42, 'unknown', { correct: 'order' });
+info('Missing placeholder is fine %s');
```

Comment:
I've allowed strings with missing template data but if we want to be more strict and require that they add it we can just change the type of `args` from `...args: ParseLogFnArgs<TMsg> | []` to `...args: ParseLogFnArgs<TMsg>` .

But, this is technically valid